Blankmon/src/application/controllers/ClickQueue.java

```diff
@@ -0,0 +1,72 @@
+package application.controllers;
+
+public class ClickQueue
+{
+	private class Node
+	{
+		private Runnable mData;
+		private Node mNext;
+
+		public Node(Runnable toRun)
+		{
+			this.mData = toRun;
+			this.mNext = null;
+		}
+	}
+
+	private Node mHead, mTail;
+	private int mSize;
+	
+	public ClickQueue()
+	{
+		mHead = null;
+		mTail = null;
+		mSize = 0;
+	}
+
+	public boolean isEmpty()
+	{
+		if((mHead == null) && (mTail == null) || mSize == 0)
+			return true;
+
+		return false;
+	}
+	
+	public void emptyQueue()
+	{
+		mHead = null;
+		mTail = null;
+		mSize = 0;
+	}
+
+	public void enqueue(Runnable val)
+	{
+		Node node = new Node(val);
+		
+		if(mSize == 0)
+			mHead = node;
+		
+		else
+			mTail.mNext = node;
+
+		mTail = node;
+		mSize++;
+	}
+
+	public Runnable dequeue()
+	{
+		if(mHead == null)
+			return null;
+		
+		Runnable val;
+		
+		val = mHead.mData;
+		mHead = mHead.mNext;
+		mSize--;
+		
+		if(mSize == 0)
+			mTail = null;
+		
+		return val;
+	}
+}
```

Comment:
Hey, could you document why you made this class and share it with the team at the next meeting? I would love to hear why personally. Sorry if it is boring for everyone else. 😄

Reply:
Sure, though the long story short is that, like in the pokemon games, I want each event to only accrue after the player has hit a button (or in this case, clicked the mouse). That way it goes like this: Enemy attacked, wait till click, player attacked, wait till clicked, prompt for next turn. Makes it so that the user can properly understand what is happening along with ensure the animations can complete.